Add templates for Web Extensions page types (#42204)
mdn/content: files/en-us/mdn/writing_guidelines/page_structures/page_types/webextension_api_property_page_template/index.md

Reviewer: +1 on this. See also: https://github.com/mdn/content/blob/main/scripts/content/release-firefox.md?plain=1 which is another good candidate for a templates dir

---
title: WebExtension API property page template
slug: MDN/Writing_guidelines/Page_structures/Page_types/WebExtension_API_property_page_template
page-type: mdn-writing-guide
sidebar: mdnsidebar
---

> [!NOTE]
> _Remove this whole explanatory note before publishing._
>
> ---
>
> **Page front matter:**
>
> The front matter at the top of the page is used to define "page metadata".

> > The values should be updated appropriately for the particular WebExtension API property.

Reviewer 2: ```suggestion
> The values must be updated appropriately for the WebExtension API property.
```

> - **slug**
>   - : The end of the URL path after `https://developer.mozilla.org/en-US/docs/`.
>     This is formatted like `Mozilla/Add-ons/WebExtensions/API/Namespace/propertyName`.
> - **page-type**
>   - : The `page-type` key for WebExtension API properties is always `webextension-api-property`.
> - **browser-compat**
>   - : Replace the placeholder value `webextensions.api.Namespace.propertyName` with the query string for the property in the [Browser compat data repo](https://github.com/mdn/browser-compat-data).
>     The toolchain automatically uses the key to populate the compatibility section (replacing the `\{{Compat}}` macros).
>
>     Note that you may first need to create/update an entry for the API property in our [Browser compat data repo](https://github.com/mdn/browser-compat-data).
>     See our [guide on how to do this](/en-US/docs/MDN/Writing_guidelines/Page_structures/Compatibility_tables).
>
> ---
>
> **Top-of-page macros**
>
> A number of macro calls appear at the top of the content section (immediately below the page front matter).
>

> > These macros are automatically added by the toolchain (there is no need to add/remove):

Reviewer 2: I don't believe this information is correct for web extensions, see  https://github.com/mdn/browser-compat-data/tree/main/docs/data-guidelines#webextensions-features-do-not-have-status-blocks

> webextensions features do not have status blocks
> 
> Web extensions are not governed by any specifications and are up to the browser vendors to design and implement. As such, the status blocks would be identical for all web extensions features. Do not add status blocks to the webextensions folder.
> 
>This guideline was adopted since the beginning of the project. for example [tabs/sendRequest](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/sendRequest)

>   To generate the correct sidebar for your API, you need to add a `GroupData` entry to our GitHub repo, and include the entry's name inside the macro call in place of _GroupDataName_.
>   See our [API reference sidebars](/en-US/docs/MDN/Writing_guidelines/Howto/Write_an_api_reference/Sidebars) guide for information on how to do this.
>
> Do not provide status header macros manually. Refer to the section ["How feature statuses are added or updated"](/en-US/docs/MDN/Writing_guidelines/Page_structures/Feature_status#how_feature_statuses_are_added_or_updated) to add these statuses to the page.
>
> Samples of the **Secure context**, **Available in workers**, **Experimental**, **Deprecated**, and **Non-standard** banners are shown right after this note block.
>
> _Remember to remove this whole explanatory note before publishing._

{{SeeCompatTable}}{{Deprecated_Header}}

Begin the content on the page with an introductory paragraph — start by naming the property, saying what WebExtension namespace it is part of, and saying what it represents and how it is used.
This should ideally be one or two short sentences.

## Value

Explain the type and meaning of the property's value.

> - If the property is a simple value (for example, a string or boolean), describe the type and possible values.

Reviewer 2: ```suggestion
- If the property is a value (for example, a string or boolean), describe the type and possible values.
```
